src/rules/derived-has-same-inputs-outputs.ts

```diff
@@ -0,0 +1,110 @@
+import type * as ESTree from "estree"
+import { createRule } from "../utils"
+import type { RuleContext } from "../types"
+import { extractStoreReferences } from "./reference-helpers/svelte-store"
+
+export default createRule("derived-has-same-inputs-outputs", {
+  meta: {
+    docs: {
+      description: "",
+      category: "Best Practices",
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      unexpected: "The argument name should be '{{name}}'.",
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    /** check node type */
+    function isIdentifierOrArrayExpression(
+      node: ESTree.SpreadElement | ESTree.Expression,
+    ): node is ESTree.Identifier | ESTree.ArrayExpression {
+      return ["Identifier", "ArrayExpression"].includes(node.type)
+    }
+
+    type ArrowFunctionExpressionOrFunctionExpression =
+      | ESTree.ArrowFunctionExpression
+      | ESTree.FunctionExpression
+
+    /** check node type */
+    function isFunctionExpression(
+      node: ESTree.SpreadElement | ESTree.Expression,
+    ): node is ArrowFunctionExpressionOrFunctionExpression {
+      return ["ArrowFunctionExpression", "FunctionExpression"].includes(
+        node.type,
+      )
+    }
+
+    /**
+     * Check for identifier type.
+     * e.g. derived(a, ($a) => {});
+     */
+    function checkIdentifier(
+      context: RuleContext,
+      args: ESTree.Identifier,
+      fn: ArrowFunctionExpressionOrFunctionExpression,
+    ) {
+      const fnParam = fn.params[0]
+      if (fnParam.type !== "Identifier") return
+      const expectedName = `$${args.name}`
+      if (expectedName !== fnParam.name) {
+        context.report({
+          node: fn,
+          loc: {
+            start: fnParam.loc?.start ?? { line: 1, column: 0 },
+            end: fnParam.loc?.end ?? { line: 1, column: 0 },
+          },
```

Comment:
Nodes handled by ESLint always have a loc property, so non-null expressions can be used.
https://eslint.org/docs/latest/developer-guide/working-with-custom-parsers#all-nodes

```suggestion
          loc: fnParam.loc!,
```

Reply:
fixed! https://github.com/ota-meshi/eslint-plugin-svelte/commit/ba6dcccb7f8c182895f8084b2f951c4087aeaaa1
And my previous PR also should do the same way, so I updated it.